tests/Neo.UnitTests/Ledger/UT_MemoryPool.cs

```diff
@@ -241,6 +242,168 @@ public async Task BlockPersistAndReverificationWillAbandonTxAsBalanceTransfered(
+            _unit.TryAdd(mp2, engine.Snapshot);
+
+            var mp3 = CreateTransactionWithFeeAndBalanceVerify(2 * txFee);  // mp3 conflicts with mp1 and has larger network fee
+            mp3.Attributes = new TransactionAttribute[] { new Conflicts() { Hash = mp1.Hash } };
+            _unit.TryAdd(mp3, engine.Snapshot);
+
+            var mp4 = CreateTransactionWithFeeAndBalanceVerify(3 * txFee);  // mp4 conflicts with mp3 and has larger network fee
+            mp4.Attributes = new TransactionAttribute[] { new Conflicts() { Hash = mp3.Hash } };
+
+            var malicious = CreateTransactionWithFeeAndBalanceVerify(3 * txFee);  // malicious conflicts with mp3 and has larger network fee, but different sender
+            malicious.Attributes = new TransactionAttribute[] { new Conflicts() { Hash = mp3.Hash } };
+            malicious.Signers = new Signer[] { new Signer() { Account = new UInt160(Crypto.Hash160(new byte[] { 1, 2, 3 })), Scopes = WitnessScope.None } };
+
+            var mp5 = CreateTransactionWithFeeAndBalanceVerify(2 * txFee);  // mp5 conflicts with mp4 and has smaller network fee
+            mp5.Attributes = new TransactionAttribute[] { new Conflicts() { Hash = mp4.Hash } };
+
+            _unit.SortedTxCount.Should().Be(2);
+            _unit.UnverifiedSortedTxCount.Should().Be(0);
+
+            // Act & Assert: try to add conlflicting transactions to the pool.
+            _unit.TryAdd(mp1, engine.Snapshot).Should().Be(VerifyResult.HasConflicts); // mp1 conflicts with mp2 and mp3 but has lower network fee than mp3 => mp1 fails to be added
+            _unit.SortedTxCount.Should().Be(2);
+            _unit.GetVerifiedTransactions().Should().Contain(new List<Transaction>() { mp2, mp3 });
+
+            _unit.TryAdd(malicious, engine.Snapshot).Should().Be(VerifyResult.HasConflicts); // malicious conflicts with mp3, has larger network fee but malicious (different) sender => mp3 shoould be left in pool
+            _unit.SortedTxCount.Should().Be(2);
+            _unit.GetVerifiedTransactions().Should().Contain(new List<Transaction>() { mp2, mp3 });
+
+            _unit.TryAdd(mp4, engine.Snapshot).Should().Be(VerifyResult.Succeed); // mp4 conflicts with mp3 and has larger network fee => mp3 shoould be removed from pool
+            _unit.SortedTxCount.Should().Be(2);
+            _unit.GetVerifiedTransactions().Should().Contain(new List<Transaction>() { mp2, mp4 });
+
+            _unit.TryAdd(mp1, engine.Snapshot).Should().Be(VerifyResult.Succeed); // mp1 conflicts with mp2 and has same network fee => mp2 shoould be removed from pool
+            _unit.SortedTxCount.Should().Be(2);
+            _unit.GetVerifiedTransactions().Should().Contain(new List<Transaction>() { mp1, mp4 });
+
+            _unit.TryAdd(mp5, engine.Snapshot).Should().Be(VerifyResult.HasConflicts); // mp1 conflicts with mp2 and has same network fee => mp2 shoould be removed from pool
```

Comment:
This case is quite complex.
Please, check this out @superboyiii.

What would happen in this case if the order of the transactions arrives differently in different nodes?
How to ensure that `mp1` did not arrive first and is removed in favor of `mp2`?
Because that case could lead to a `primary` proposing a block that can not be obtained by some nodes due to this order of arriving conflicts.

Reply:
First of all, I've noticed that the comment to this testcase was wrong (it was duplicating, I forgot to change it while copying the testcase). The comment is fixed now.

@vncoelho,
> What would happen in this case if the order of the transactions arrives differently in different nodes?

It's OK because conflicting transactions have the *strict rule* of sorting and removing. Consider the described in the test case: `tx1` doesn't have `Conflicts` attributes; `tx2` has `Conflicts` attribute with the hash of `tx1` and has the same network fee. Imagine that `Node1` has `tx2` in the pool, after that we try to add `tx1` to the pool. `tx1` should be successfully added to the pool and `tx2` will be removed because:

- `tx1` and `tx2` have the same sender
- `tx1` and `tx2` have the same network fee
- `tx1` doesn't have `Conflicts` attribute, it's the `tx2` who has this attribute

Imagine at the same time that `Node2` has received the transactions in different order and there's `tx1` in the pool of `Node2`. After that `Node2` receives the `tx2` and fails to add it to the mempool, because:

- `tx1` and `tx2` have the same sender
- `tx1` and `tx2` have the same network fee
- but it's `tx2` who has `Conflicts` attribute, thus, `tx1` wins.

In the end we've got `tx1` in the pool of both `Node1` and `Node2`; `tx2` is missing from both pools, so the state is consistent between the nodes.

Please, check the updated testcase, I've added this situation to `TryAdd_AddRangeOfConflictingTransactions()`.

Reply:
> primary proposing a block that can not be obtained by some nodes due to this order of arriving conflicts.

It's impossible, because eventually dBFT will get all transactions of the proposed block from the P2P naighbours. And if the block is correct, then it can't contain two conflicting transactions.

Reply:
Now, and how about a similar scenario:
* tx1 had conflits* with fee 1 method 1
* tx2 conflicts tx1 with fee 10 method 2
* tx2 conflicts tx1 with fee 10 method 3

Reply:
Then tx2 and tx3 will stay in the mempool and tx1 will be kicked off, because tx1 has smaller network fee, there's a testcase for that situation in the PR:
```
            _unit.TryAdd(mp1, engine.Snapshot).Should().Be(VerifyResult.HasConflicts); // mp1 conflicts with mp2 and mp3 but has lower network fee than mp3 => mp1 fails to be added
            _unit.SortedTxCount.Should().Be(2);
            _unit.GetVerifiedTransactions().Should().Contain(new List<Transaction>() { mp2, mp3 });
```
From two conflicting transactions wins the one that has larger network fee. If they have the same network fee then wins the one that doesn't have Conflicts attribute with the hash of another one (i.e. wins the one that was created earlier than another).

Reply:
Both tx2 and tx3 have the same fees and conflicts with 1.
They will both stay in the mempool,right?

Reply:
Yes, that’s right.